lib/elixir/src/elixir_parser.yrl

```diff
@@ -256,6 +258,13 @@ access_expr -> bit_string : '$1'.
 access_expr -> sigil : build_sigil('$1').
 access_expr -> max_expr : '$1'.
 
+%% Numbers in different bases
+number -> binary : handle_literal(?exprs('$1'), '$1', {base, 2}).
+number -> octal : handle_literal(?exprs('$1'), '$1', {base, 8}).
+number -> decimal : handle_literal(?exprs('$1'), '$1', {base, 10}).
+number -> float : handle_literal(?exprs('$1'), '$1').
+number -> hex : handle_literal(?exprs('$1'), '$1', {base, 16}).
```

Comment:
I don't want to call it `{base, 16}` because we also need to handle `char`, such as `?é`, which also simply become integers.

My suggestion is to call it format:

```erlang
number -> char : handle_literal(?exprs('$1'), '$1', [{format, char}]).
number -> binary : handle_literal(?exprs('$1'), '$1', [{format, binary}]).
number -> octal : handle_literal(?exprs('$1'), '$1', [{format, octal}]).
number -> decimal : handle_literal(?exprs('$1'), '$1', [{format, decimal}]).
number -> hex : handle_literal(?exprs('$1'), '$1', [{format, hexadecimal}]).
number -> float : handle_literal(?exprs('$1'), '$1').
```

Then we need to also remove `char` from here: https://github.com/elixir-lang/elixir/pull/6344/files#diff-dbe798b7d4bb8070f478211481ccf34aR248